py/plugins/google-genai/src/genkit/plugins/google_genai/google.py

```diff
@@ -48,41 +49,94 @@ def google_genai_name(name: str) -> str:
     Returns:
         The fully qualified Google AI action name.
     """
-    return f'{PLUGIN_NAME}/{name}'
+    return f'{GOOGLEAI_PLUGIN_NAME}/{name}'
 
 
-class GoogleGenai(Plugin):
+def google_vertexai_name(name: str) -> str:
+    """Create a Google AI action name.
+
+    Args:
+        name: Base name for the action.
+
+    Returns:
+        The fully qualified Google AI action name.
+    """
+    return f'{VERTEXAI_PLUGIN_NAME}/{name}'
+
+
+class GoogleAI(Plugin):
     """Google Ai plugin for Genkit"""
 
-    name = PLUGIN_NAME
+    name = GOOGLEAI_PLUGIN_NAME
+    _vertexai = False
 
     def __init__(
         self,
-        vertexai: bool | None = None,
         api_key: str | None = None,
         credentials: Credentials | None = None,
-        project: str | None = None,
-        location: str | None = None,
         debug_config: DebugConfig | None = None,
         http_options: HttpOptions | HttpOptionsDict | None = None,
     ):
-        """Initialize the GoogleGenai plugin.
+        """Initialize the GoogleAI plugin.
 
         Args:
-            vertexai: Whether to use Vertex AI.
             api_key: The API key to use.
             credentials: The credentials to use.
         """
         api_key = api_key if api_key else os.getenv('GEMINI_API_KEY')
-        if not vertexai and not api_key:
+        if not api_key:
             raise ValueError(
                 'Gemini api key should be passed in plugin params or as a GEMINI_API_KEY environment variable'
             )
-        self._vertexai = vertexai
 
         self._client = genai.client.Client(
-            vertexai=vertexai,
-            api_key=api_key if not vertexai else None,
+            vertexai=self._vertexai,
+            api_key=api_key,
+            credentials=credentials,
+            debug_config=debug_config,
+            http_options=_inject_attribution_headers(http_options),
+        )
+
+    def initialize(self, ai: GenkitRegistry) -> None:
+        """Initialize the plugin by registering actions in the registry.
+
+        Args:
+            ai: the action registry.
+
+        Returns:
+            None
+        """
+        supported_gemini_models = list(GeminiVersion) + list(GeminiApiOnlyVersion)
```

Comment:
probably would cleaner to have `VertexGeminiVersions` and `GoogleAiGeminiVersions`?